api/envoy/api/v2/core/protocol.proto

```diff
@@ -50,6 +62,11 @@ message Http1ProtocolOptions {
   // Envoy does not otherwise support HTTP/1.0 without a Host header.
   // This is a no-op if *accept_http_10* is not true.
   string default_host_for_http_10 = 3;
+
+  // Describes how the keys for response headers should be formatted. By default, all header keys
+  // are lower cased.
+  // [#not-implemented-hide:] Hiding until both request and response formating has been implemented.
```

Comment:
Any reason to not just do it here?  Is it more work than creating the formatter on the client codec?
(I'm totally fine with a follow-up if you prefer)

Reply:
Adding the codec stuff is easy, the thing i was trying to avoid was doing all the plumbing + tests to make the h1 protocol options available to the connection pool. I'll do a fast follow once this lands